src/util/string_expr.h

```diff
@@ -182,7 +182,7 @@ inline void validate_expr(const refined_string_exprt &x)
 {
   INVARIANT(x.id() == ID_struct, "refined string exprs are struct");
   validate_operands(x, 2, "refined string expr has length and content fields");
-  INVARIANT(x.length().type().id() == ID_signedbv, "length is an unsigned int");
+  INVARIANT(x.length().type().id() == ID_signedbv, "length is a signed int");
```

Comment:
:pick: Consider changing this to `length should be a signed int`. Now it is not quite clear whether the length is or should be signed.